src/qt/receivecoinsdialog.cpp

```diff
@@ -95,6 +101,8 @@ ReceiveCoinsDialog::~ReceiveCoinsDialog()
 void ReceiveCoinsDialog::clear()
 {
     ui->reqAmount->clear();
+    address = getAddress();
```

Comment:
after creating a new payment request, reusing an address, this method is called when  `ui->reuseAddress` is still checked. Thus the `getAddress()` method opens a new dialog right after closing the previous one.

Besides, a new address shouldn't necessarily be generated here imho.